task8/js/app.js

```diff
@@ -0,0 +1,155 @@
+// 1. Use the window.location object to redirect the user
+// to a different webpage when a button is clicked.
+
+function redirectToGoogle() {
+  window.location.href = "https://www.google.com";
+}
+
+// 2. Use the window.history object to go back to the previous page when a button is clicked.
+
+function goBackToPreviousPage() {
+  window.history.back();
+}
+
+// 3. Access and display the user's browser name and version using the window.navigator object.
+
+function displayBrowserInfo() {
+  const browserUserAgent = window.navigator.userAgent;
+  const arr = [
+    ["Microsoft Edge", /edg/i],
+    ["Microsoft Internet Explorer", /trident/i],
+    ["Mozilla Firefox", /firefox|fxios/i],
+    ["Opera", /opr\//i],
+    ["UC Browser", /ucbrowser/i],
+    ["Samsung Browser", /samsungbrowser/i],
+    ["Google Chrome", /chrome|chromium|crios/i],
+    ["Apple Safari", /safari/i],
+    ["Unknown", /.+/i],
+  ];
+  let br = "Unknown";
+  let num = "unknown";
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i][1].test(browserUserAgent)) {
+      br = arr[i][0];
+      num = browserUserAgent.split(arr[i][1])[1].split(" ")[0].replace("/", "");
+      break;
+    }
+  }
```

Comment:
Array.find() ?
From "best practice" point of view loops like `for` is used in very specific cases.